prefab_cloud_python/__init__.py

```diff
@@ -5,3 +5,47 @@
 from ._internal_setup import create_prefab_structlog_processor
 from ._internal_setup import default_structlog_setup
 from .constants import STRUCTLOG_CALLSITE_IGNORES
+from importlib.metadata import version
+from .read_write_lock import ReadWriteLock
+import logging
+
+
+__base_client = None
+__options = None
+__lock = ReadWriteLock()
+
+
+def set_options(options: Options) -> None:
+    global __base_client
+    global __options
+    global __lock
```

Comment:
uber-nit: I don't think you have to explicitly declare these global variables here ([context](https://docs.python.org/3/faq/programming.html#what-are-the-rules-for-local-and-global-variables-in-python))